src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public boolean equals(Object figure) {
+        if (this == figure) {
+            return true;
+        }
+        if (figure == null) {
+            return false;
+        }
+        if (figure.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) figure;
+            return color.equals(rectangle.color)
+                    && width.equals(rectangle.width)
+                    && length.equals(rectangle.length);
+        }
+
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        result = 31 * result + width.hashCode();
+        result = 31 * result + length.hashCode();
```

Comment:
check for null before calling "field.hashcode()", cause you can get nullPointer in some cases, it's better to avoid it